test/sample-test/Dockerfile

```diff
@@ -11,6 +11,10 @@ RUN wget https://bootstrap.pypa.io/get-pip.py && python3 get-pip.py
 COPY ./test/sample-test/requirements.txt /python/src/github.com/kubeflow/pipelines/test/sample-test/requirements.txt
 RUN pip3 install -r /python/src/github.com/kubeflow/pipelines/test/sample-test/requirements.txt
 
+# Install KFP server API from commit.
+COPY ./backend/api/python_http_client /python_http_client
```

Comment:
I'm not sure this approach is correct. This is not how the end users are installing the kfp package.
The changes in this PR could still have been submitted without these lines. They'd just have to be split into two granular PRs which is usually a good idea.

Reply:
Without this line the e-2-e testing would fail as there would be a mismatch between the kfp client and kfp-api-server.

Reply:
That's the goal of the e2e tests - to detect when things are going to fail for our users.
Here we're essentially disabling part of the testing by performing manual installation.

A better approach might have been to make atomic PRs:

1. Add new API to backend
2. Expose API in the generated clients
3. Use the clients in the SDKs

Reply:
This was discussed here: https://github.com/kubeflow/pipelines/pull/4638#issuecomment-729620679. Seeing as the kfp-server-api and the SDK depend on each other, when installing from master they should both be installed from master.